toolkit/tools/pkgworker/pkgworker.go

```diff
@@ -214,34 +215,39 @@ func buildSRPMInChroot(chrootDir, rpmDirPath, toolchainDirPath, workerTar, srpmF
 
 	err = chroot.Initialize(workerTar, extraDirs, mountPoints, true)
 	if err != nil {
+		err = fmt.Errorf("failed to initialize chroot:\n%w", err)
 		return
 	}
 	defer chroot.Close(noCleanup)
 
 	// Place extra files that will be needed to build into the chroot
 	srpmFileInChroot, err := copyFilesIntoChroot(chroot, srpmFile, repoFile, rpmmacrosFile, runCheck)
 	if err != nil {
+		err = fmt.Errorf("failed to copy files into chroot:\n%w", err)
 		return
 	}
 
 	// Run the build in a go routine so we can monitor and kill it if it takes too long.
 	results := make(chan error)
-	go func() {
-		buildErr := chroot.Run(func() (err error) {
-			return buildRPMFromSRPMInChroot(srpmFileInChroot, outArch, runCheck, defines, packagesToInstall, isCCacheEnabled(ccacheManager))
-		})
-		results <- buildErr
-	}()
-
-	select {
-	case err = <-results:
-	case <-time.After(timeout):
-		logger.Log.Errorf("Timeout after %v: killing all processes in chroot...", timeout)
-		shell.PermanentlyStopAllChildProcesses(unix.SIGKILL)
-		err = fmt.Errorf("build timed out after %s", timeout)
-	}
+	err = chroot.Run(func() (err error) {
+		go func() {
+			results <- buildRPMFromSRPMInChroot(srpmFileInChroot, outArch, runCheck, defines, packagesToInstall, isCCacheEnabled(ccacheManager))
+		}()
+
+		var chrootErr error = nil
+		select {
```

Comment:
Why was this moved under `chroot.Run`?

Reply:
Added a quick blurb to the description, but more details:

So, we rely on `chroot.Run()` to both setup and tear down the chroot (ie restore the root dir). If we do the timeout outside the `chroot.Run()`, we will exit the function and begin teardown immediately. Unfortunately... we won't have restored the correct root yet, so all the cleanup code will be running relative to the chroot's root dir.

This means we will try to unmount stuff, but the path doesn't actually exist, so it fails.

Reply:
It would be good to add a comment in the code.

Reply:
Done